app/views/software_records/_form_server_environment.html.erb

```diff
@@ -0,0 +1,80 @@
+            <div class="form-group">
+              <%= form.label :qa_url %>
+              <%= form.text_area :qa_url, :class => "form-control" %>
+            </div>
+
+            <div class="form-group">
+              <%= form.label :dev_url %>
+              <%= form.text_area :dev_url, :class => "form-control" %>
+            </div>
+
+            <div class="form-group">
+              <%= form.label :production_server %>
+              <%= form.text_area :production_server, :class => "form-control" %>
+            </div>
+
+            <div class="form-group">
+              <%= form.label :production_support_servers %>
+              <%= form.text_area :production_support_servers, :class => "form-control" %>
+            </div>
+
+            <div class="form-group">
+              <%= form.label "last_record_change" %>
+               <%= form.date_field :last_record_change, as: :date, value: form.object.try(:strftime,"%m/%d/%Y"), class: 'form-control' %>
+            </div>
+
+            <% if params[:id].nil? %>
+              <div class="form-group">
+                <%= form.label :track_uptime, 'Track uptime?' %><br/>
```

Comment:
I want to make all of these "Yes|No" values boolean data types and switch to checkboxes. I just now noticed that our current pattern uses the controller to help track the values and I don't think it's sustainable.

Reply:
Use radio buttons with yes/no.   

Should some of that controller logic been in the model?